### ISSUE TITLE
Regression: Value 'Collection' is unsubscriptable 

### ISSUE THREAD

**Issue by Spacerat** [PyCQA/pylint#2377]
### Steps to reproduce
Run pylint on this file

```py
from typing import Collection

def myfunc(arg: Collection[int]):
    print(arg)

myfunc([1, 2, 3])
```

### Current behavior

`repro.py:3:16: E1136: Value 'Collection' is unsubscriptable (unsubscriptable-object)`

### Expected behavior

No errors.

### pylint --version output

Breaks for 2.1.0:
```
pylint 2.1.0
astroid 2.0.2
Python 3.6.4 (default, Mar 12 2018, 17:50:21)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

Does not break for 1.9.2:
```
pylint 1.9.2,
astroid 1.6.5
Python 3.6.4 (default, Mar 12 2018, 17:50:21)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

If there is a configuration workaround for this (except for disabling `unsubscriptable-object` globally) that would be nice, but neither `ignored-modules=typing` or `ignored-classes=Collection` worked for me.

**Comment by Spacerat** [comment]
In fact, this seems to be a Python 3.6 specific issue.
```
pylint 2.1.0
astroid 2.0.2
Python 3.7.0 (default, Aug  1 2018, 18:01:40)
[Clang 9.0.0 (clang-900.0.39.2)]
```
There's no issue for 3.7, so I'm not sure if this should be considered a regression

**Comment by mthuurne** [comment]
This issue also happens for me with the following versions:
```
pylint 2.2.2
astroid 2.0.4
Python 3.6.5 (default, Mar 31 2018, 19:45:04) [GCC]
```

**Comment by mplanchard** [comment]
Also seeing this with

```raw
pylint 2.3.1
astroid 2.2.5
Python 3.6.8 (default, Mar  5 2019, 06:26:06)
```

**Comment by mthuurne** [comment]
Looking at the code of `typing.py`, the way `Collection` is defined is very different between Python 3.6 and 3.7, so fixing this might not be trivial.

I think `ignored-classes=Collection` does not help as a workaround because PyLint doesn't consider `Collection` a class on Python 3.6: it reports `Value 'Collection' is unsubscriptable` after all.

I ended up using this workaround:
```py
from typing import TYPE_CHECKING, Collection

if not TYPE_CHECKING:
    # Work around PyLint issue on Python 3.6.
    #   https://github.com/PyCQA/pylint/issues/2377
    from typing import Sequence # pylint: disable=ungrouped-imports
    Collection = Sequence # pylint: disable=invalid-name
```

**Comment by tuukkamustonen** [comment]
Also occurs on python 3.5:

```
pylint 2.4.4
astroid 2.3.1
Python 3.5.5 (default, May 29 2018, 10:15:24) 
[GCC 5.4.0 20160609]
```

**Comment by jnsnow** [comment]
I'm working on adding pylint (+mypy) to all of QEMU's CI for our test infrastructure. We use Python 3.6 as our minimum viable python version for test infrastructure, so we are encountering this issue.

It appears to be an issue on all 2.x.x pylint versions against Python 3.6.10.

For now, my workaround will likely be just using `Sequence` instead of `Collection`.

**Comment by remort** [comment]
Still hurts on
```
pylint 2.6.0
astroid 2.4.2
Python 3.6.8 (default, Jan 30 2019, 23:58:16) 
```

**Comment by cdce8p** [comment]
This issue seems to be fixed when testing with latest pylint version. Closing it.
If you still encounter it, please report back and I'll be happy to reopen it.
